provider/adapters/src/any_payload.rs

```diff
@@ -47,6 +40,31 @@ pub struct AnyPayloadProvider {
     pub data: AnyPayload,
 }
 
+impl AnyPayloadProvider {
+    pub fn new_owned<M: ResourceMarker + 'static>(data: M::Yokeable) -> Self {
```

Comment:
Nit: Add docs, please. If this crate doesn't already have the docs lint enabled, that's a separate issue.